pkg/discovery/dtofactory/pod_entity_dto_builder.go

```diff
@@ -596,6 +599,30 @@ func (builder *podEntityDTOBuilder) createContainerPodData(pod *api.Pod) *proto.
 	if util.PodIsReady(pod) && pod.Status.PodIP == "" {
 		glog.Errorf("No IP found for pod %s", fullName)
 	}
+
+	ownerInfo, err := util.GetPodParentInfo(pod)
+	if err != nil || util.IsOwnerInfoEmpty(ownerInfo) {
+		// Pod does not have controller
+		glog.V(3).Infof("Skip adding workload controller data for pod %v/%v: pod has no controller.",
+			pod.Namespace, pod.Name)
+		return podData
+	}
+
+	controller, exists := builder.ClusterSummary.ControllerMap[ownerInfo.Uid]
+	if !exists {
+		// Could be custom controller. We do not bulk process custom controller.
+		glog.V(3).Infof("add cumtomer controller data of controller %v/%v for pod %v/%v: controller not cached.",
```

Comment:
Does this mean that we might update the type as custom controller for a replicaset or deployment parent if the same has not been cached yet??
Would using `GetPodControllerInfo()` (currently a method on clusterScraper) be better in this case.

Reply:
I changed the code to use GetPodControllerInfo() to get the controller type for the pod. Please check that out and help me review on the default case of CustomControllerData. Thanks for the help on this again!